### PR TITLE
Add error message when the payment method is invalid in Back Office's Orders page

### LINKED ISSUE #22587: BO - Order page - Unclear alert when adding a payment with invalid payment method
#### Describe the bug

In the BO > Orders > Order view page > when we add a payment with invalid payment => an Unclear error is displayed 

#### Steps to Reproduce

1. Go to BO > Orders > Old order page
2. Add payment with "Payment method" `Paym>>ents by check`
3. Add
4. See an error

image

PS1769 => ok
image

#### Additional information

* PrestaShop version: 1770, 1771build1
* PHP version: 7.0

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.7.x
| Description?      | BO - Order Page - Set an error message when the payment method is invalid
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #22587
| How to test?      | 1. Go to BO > Orders > Old order page<br>2. Add payment with "Payment method" Paym>>ents by check<br>3.  Click on Add<br>4. **BEFORE** See an error<br>4. **AFTER** An error message is displayed
| Possible impacts? | 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22791)

### CODE REVIEW

**Review by matks — DISMISSED** [review]

**Review by Arman-Hosseini — DISMISSED** [review]

**Comment by Julievrz** [comment]
Hi @Progi1984, what would you think of adding the list of invalid characters in the error message?

I'm not sure which characters are invalid though, because I tried # ; - _ and it worked. 

I would say : 'Invalid characters: <>={}"

**Comment by Progi1984** [comment]
> Hi @Progi1984, what would you think of adding the list of invalid characters in the error message?
> 
> I'm not sure which characters are invalid though, because I tried # ; - _ and it worked.
> 
> I would say : 'Invalid characters: <>={}"

@Julievrz Invalid characters are : `^<>={}`

**Comment by Julievrz** [comment]
Great, so we could use the string you suggested ("The selected payment method is invalid." in Admin.OrdersCustomers.Notification) following by this string: "Invalid characters:" in Admin.Notification.Info

**Comment by Progi1984** [comment]
@Julievrz Done.

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 1]
```diff
@@ -1926,6 +1926,15 @@ private function getPaymentErrorMessages(Exception $e)
                 'Only numbers and decimal points (".") are allowed in the amount fields of the payment block, e.g. 10.50 or 1050.',
                 'Admin.Orderscustomers.Notification'
             ),
+            OrderConstraintException::class => [
```
Comment (matthieu-rolland) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php:1929]: Maybe using php's `sprintf` method would be more readable ?

Sorry for the indentation, but something like this:

```php
OrderConstraintException::class => [
    OrderConstraintException::INVALID_PAYMENT_METHOD =>
        sprintf(
            '%s %s ^<>={}',
            $this->trans(
                'The selected payment method is invalid.',
                'Admin.Orderscustomers.Notification'
            ),
            $this->trans(
                'Invalid characters:',
                'Admin.Notifications.Info'
            )
        )
],

```

Comment (jolelievre) [reply src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php:1929]: It would be nicer indeed

**Review by matthieu-rolland — APPROVED** [review]

**Review by jolelievre — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

I found this issue:

The alert displayed is `The selected payment method is invalid. Invalid characters: ^<>={}`

1. Go to BO > Orders > Old order page
2. Add payment with "Payment method" `Payme^nts b^y che^ck`
3. Add
4. The payment well added

image

So, this symbol `^` is not an invalid character, I think, it should be removed from the alert.

I tried with PS1769 => ok
image


Thank you!

**Comment by Progi1984** [comment]
@khouloudbelguith It seems that it has been added in 177x by @sarjon in #13712.

@PrestaShop/product-team Do i remove the `^` from invalid characters ? or I let it ?

**Review by PierreRambaud — DISMISSED** [review]

**Comment by MatShir** [comment]
You can remove the invalid character constraint 👍

**Review by matthieu-rolland — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

I found an exception following these steps:

1. Go to BO > Orders > Old order page
2. Add payment with "Payment method" only space:  ` `

3. Click on Add
4. See an error

image

This issue is reproduced with & without PR.

In the PS1769 => ok, the payment well added without exceptions 
image

Ping @Progi1984 @PrestaShop/product-team what do you think?
This issue should be fixed in this PR or in another one?

Thank you!

**Comment by Progi1984** [comment]
@khouloudbelguith If "This issue is reproduced with & without PR.", could you create a new issue ? It shouldn't be fixed in this PR.

**Review by sowbiba — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1dRYHSMWqctjct6z58jH7MbN_tBUmkW69/view?usp=sharing

Thank you!

**Comment by Progi1984** [comment]
## DONT MERGE IT BEFORE 1772 RELEASE

**Comment by PierreRambaud** [comment]
Thank you @Progi1984

**Pull request merged**